Animations/Animation.js

```diff
@@ -37,20 +22,36 @@ var makeTransition = function (displayName, transitions) {
       // componentDidEnter FUNCTION: use to define actions to run after an animation has entry completed
       // componentDidLeave FUNCTION: use to define actions to run after an animation has leave completed
   return React.createClass({
+    // setAnimationDomNode FUNCTION: function to handle manipulating and setting a TransitionGroup in the DOM
+      //@param action STRING: the transition key you are looking to affect
+      //@param callback FUNCTION: callback attribute passed in from the containing function to act on function completion
+    setAnimationDomNode: function (action, callback) {
+      this.el = this.getDOMNode();
+      this.$el = $(this.el);
+
+      // requestAnimationFrame FUNCTION: Calls the specified function updating an animation before the next browser repaint. Defined in window
+      requestAnimationFrame(function () {
+        this.$el.css(transitions[action]);
+        requestAnimationFrame(function () {
+          this.$el.css(transitions[action + 'Active']);
+          Arrival(this.$el, callback);
+        }.bind(this));
+      }.bind(this));
+    },
     componentWillEnter: function (callback) {
-      setAnimationDomNode('enter', callback);
+      this.setAnimationDomNode('enter', callback);
     },
 
     componentDidEnter: function () {
-      // This is called after the callback function that was passed to componentWillEnter is called.
+      // This is called after the callback function that was passed to componentWillEnter is called
     },
 
     componentWillLeave: function (callback) {
-      setAnimationDomNode('leave', callback);
+      this.setAnimationDomNode('leave', callback);
     },
 
     componentDidLeave: function () {
```

Comment:
I would prefer to have it left in with the addition of 'this.setAnimationDomNode('left', callback)'. Provided the callback is something that is passed in and should be passed into setAnimationDomNode. Either that or it should be removed, as we shouldn't be committing commented out code just in case we need it later.